src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -2020,6 +2180,260 @@ bool DiscoveryDataBase::add_edp_subscriptions_to_send_(
     return false;
 }
 
+void DiscoveryDataBase::to_json(nlohmann::json& j) const
+{
+    // participants
+    auto pit = participants_.begin();
+    j["participants"] = nlohmann::json({});
+    while(pit != participants_.end())
+    {
+        if (pit->first != server_guid_prefix_)
+        {
+            nlohmann::json j_part;
+            pit->second.to_json(j_part);
+            j["participants"][ddb::object_to_string(pit->first)] = j_part;
+        }
+        ++pit;
+    }
+
+    // writers
+    auto wit = writers_.begin();
+    if (wit == writers_.end())
+    {
+        j["writers"] = nlohmann::json({});
+    }
+    while(wit != writers_.end())
+    {
+        nlohmann::json j_w;
+        wit->second.to_json(j_w);
+        j["writers"][ddb::object_to_string(wit->first)] = j_w;
+        ++wit;
+    }
+
+    // readers
+    auto rit = readers_.begin();
+    if (rit == readers_.end())
+    {
+        j["readers"] = nlohmann::json({});
+    }
+    while(rit != readers_.end())
+    {
+        nlohmann::json j_r;
+        rit->second.to_json(j_r);
+        j["readers"][ddb::object_to_string(rit->first)] = j_r;
+        ++rit;
+    }
+
+    // TODO add version
+}
+
+
+bool DiscoveryDataBase::from_json(
+        nlohmann::json& j,
+        std::map<eprosima::fastrtps::rtps::InstanceHandle_t, fastrtps::rtps::CacheChange_t*>& changes_map)
+{   
+    // This function will parse each attribute in json backup, casting it to istringstream
+    // (std::istringstream) j[""] >> obj;
+
+    // Changes are taken from changes_map, with already created changes
+
+    // Auxiliar variables to deserialize and create new objects of the ddb
+    fastrtps::rtps::InstanceHandle_t instance_handle_aux;
+    fastrtps::rtps::GuidPrefix_t prefix_aux;
+    fastrtps::rtps::GuidPrefix_t prefix_aux_ack;
+    fastrtps::rtps::GUID_t guid_aux;
+
+    logInfo(DISCOVERY_DATABASE, "Raising DDB from json Backup");
+
+    try
+    {
+        // Participants
+        for (auto it = j["participants"].begin(); it != j["participants"].end(); ++it)
+        {
+            // Populate info from participant to charge its change
+            (std::istringstream) it.key() >> prefix_aux;
```

Comment:
In general, I would use the function notation (`std::istringstream(x)`) rather than the c-like cast notation (`(std::istringstream) x`)